providers/implementations/ciphers/cipher_aes_gcm_hw.c

```diff
@@ -59,11 +59,76 @@ static int generic_aes_gcm_initkey(PROV_GCM_CTX *ctx, const unsigned char *key,
     return 1;
 }
 
+static int generic_aes_cipher_update(PROV_GCM_CTX *ctx, const unsigned char *in,
```

Comment:
Should it be called generic_aes_gcm_cipher_update? And is the generic_ prefix necessary?

Reply:
aes_gcm_cipher_update seems like a reasonable name to me..

Reply:
That seems inconsistent, considering `generic_aes_gcm_initkey`.  Should that one be renamed as well?

Reply:
It is only used by this file - so maybe generic was a bad name.
All the hw platforms have there own name e.g- s390x_aes_gcm_initkey
This is the default name if there is no hw or if the hw is not capable..